internal/backends/selector/lib.rs

```diff
@@ -48,6 +50,60 @@ cfg_if::cfg_if! {
     }
 }
 
+pub struct PlatformBuilder {
+    opengl_api: Option<OpenGLAPI>,
+    renderer: Option<SlintRenderer>,
+}
+
+impl PlatformBuilder {
+    /// Creates a new PlatformBuilder for configuring aspects of the Platform.
+    pub fn new() -> PlatformBuilder {
+        PlatformBuilder { opengl_api: None, renderer: None }
+    }
+
+    /// Configures this builder to use the specified OpenGL API when building the platform later.
+    pub fn with_opengl_api(mut self, opengl_api: OpenGLAPI) -> Self {
+        self.opengl_api = Some(opengl_api);
+        self
+    }
+
+    /// Configures this builder to use the specified renderer when building the platform later.
+    pub fn with_renderer(mut self, renderer: SlintRenderer) -> Self {
```

Comment:
One issue with this approach: it's currently winit specific. Ideas here? Leave it for a future PR after Qt/KMS support is added?

Reply:
I think it's okay to keep it.